# HG changeset patch
# User alta88@fixall.com
# Date 1746109868 21600
# Parent  a54dea48b31a976520770650faa1348cee0de3b6
Deprecate ATOM_03
- Pref rss.ATOM_03.disabled set to true for a cycle.

diff --git a/mailnews/extensions/newsblog/FeedParser.sys.mjs b/mailnews/extensions/newsblog/FeedParser.sys.mjs
--- a/mailnews/extensions/newsblog/FeedParser.sys.mjs
+++ b/mailnews/extensions/newsblog/FeedParser.sys.mjs
@@ -566,8 +566,13 @@ FeedParser.prototype = {
   },
 
   // TODO: deprecate ATOM_03_NS.
   parseAsAtom(aFeed, aDOM) {
+    if (Services.prefs.getBoolPref("rss.ATOM_03.disabled", true)) {
+      aFeed.addLog("error", "parseAsAtom: ATOM_03 is not supported");
+      return [];
+    }
+
     // Get the first channel (assuming there is only one per Atom File).
     const channel = aDOM.querySelector("feed");
     if (!channel) {
       aFeed.onParseError(aFeed);
diff --git a/mailnews/mailnews.js b/mailnews/mailnews.js
--- a/mailnews/mailnews.js
+++ b/mailnews/mailnews.js
@@ -776,8 +776,11 @@ pref("rss.show.favicons", true);
 
 // Learn More link for feeds.
 pref("rss.learnmore.url", "https://www.betterbird.eu/expert-tips/#feed-enhancements");
 
+// Deprecate ATOM_03 spec support.
+pref("rss.ATOM_03.disabled", true);
+
 // Feed syntax validator query link.
 pref("rss.validator.url", "https://validator.w3.org/feed/check.cgi?url=");
 
 pref("feeds.loglevel", "Warn");
